client/client.go

```diff
@@ -44,54 +45,52 @@ type Client struct {
 // It establishes a connection to the blockchain and verifies the integrity of contracts at the given address.
 // It uses the comm backend to initialize adapters for off-chain communication network.
 func NewEthereumPaymentClient(cfg Config, user perun.User, comm perun.CommBackend) (*Client, error) {
-	dialer, listener, err := initComm(comm, user.CommAddr)
-	if err != nil {
-		return nil, err
-	}
 	funder, adjudicator, err := connectToChain(cfg.Chain, user.OnChain)
 	if err != nil {
 		return nil, err
 	}
-	// Only off-chain account is unlocked. Accounts for the participant addresses
-	// will be unlocked by the client when required.
 	offChainAcc, err := user.OffChain.Wallet.Unlock(user.OffChain.Addr)
 	if err != nil {
 		return nil, errors.WithMessage(err, "off-chain account")
 	}
+	msgBus := net.NewBus(offChainAcc, comm.NewDialer())
```

Comment:
Yes, you are right. I am preparing this in another task. 

I am planning to define a `Registerer` interface and assign the `dialer` to a variable of type `Registerer` in `client`. So that we can stored any dialer implementation.